src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Util.java

```diff
@@ -87,14 +87,19 @@ private static Provider getProvider(Provider p, String providerName,
         }
         p = Security.getProvider(providerName);
         if (p == null) {
-            try {
-                @SuppressWarnings("deprecation")
-                Object o = Class.forName(className).newInstance();
-                p = (Provider)o;
-            } catch (Exception e) {
-                throw new ProviderException
-                        ("Could not find provider " + providerName, e);
-            }
+            p = AccessController.doPrivileged(
```

Comment:
Could you use the limited version of doPrivileged and only assert the permissions that are strictly necessary to instantiate a provider?

Reply:
Yes, makes sense. Thanks for your feedback.